### PR TITLE
[INLONG-6785][Manager] Support register and manage the resource of Apache Hudi

### LINKED ISSUE #6785: [Feature][Manager] Support register and manage the resource of Apache Hudi
### Description

[Manager] Support register and manage hudi sort

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ZuoFengZhang**
### Prepare a Pull Request
*[Manager] Support register and manage hudi sort*

- Fixes #6785 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [x] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/hudi/HudiSink.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import org.apache.inlong.manager.common.consts.SinkType;
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.common.util.JsonTypeDefine;
+import org.apache.inlong.manager.pojo.sink.SinkRequest;
+import org.apache.inlong.manager.pojo.sink.StreamSink;
+
+/**
+ * Iceberg sink info
```
Comment (leosanqing) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/hudi/HudiSink.java:37]: Hudi

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartStreamListener.java [hunk 2]
```diff
@@ -38,12 +43,6 @@
 import org.apache.inlong.manager.workflow.event.ListenerResult;
 import org.apache.inlong.manager.workflow.event.task.SortOperateListener;
 
-import java.util.HashMap;
```
[the remark this answers is not in the document]
Comment (ZuoFengZhang) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartStreamListener.java:41]: Ok, I have a question why the spotless script doesn't work

[the remark this answers is not in the document]
Comment (dockerzhang) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartStreamListener.java:41]: > Please refer this: #3082, and reformat those packages' order, thanks.

spotless support defines the order of the import packages.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/hudi/HudiPartition.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.pojo.sink.hudi;
+
+import org.apache.inlong.manager.common.util.Preconditions;
+
+/**
+ * Iceberg partition type
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/hudi/HudiPartition.java:23]: Hudi

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+import org.apache.hudi.org.apache.hbase.thirdparty.com.google.common.collect.Maps;
+import org.apache.hudi.sync.common.util.ConfigUtils;
+import org.apache.hudi.table.catalog.HiveSchemaUtils;
+import org.apache.inlong.manager.pojo.sink.hudi.HudiColumnInfo;
+import org.apache.inlong.manager.pojo.sink.hudi.HudiTableInfo;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HudiCatalogClient {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java:54]: Please add Java docs for classes, public methods, thanks.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartStreamListener.java [hunk 5]
```diff
@@ -89,9 +89,11 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
 
         Map<String, String> kvConf = new HashMap<>();
         groupExtList.forEach(groupExtInfo -> kvConf.put(groupExtInfo.getKeyName(), groupExtInfo.getKeyValue()));
-        streamExtList.forEach(extInfo -> {
-            kvConf.put(extInfo.getKeyName(), extInfo.getKeyValue());
-        });
+        if (streamExtList != null) {
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartStreamListener.java:92]: // import org.apache.commons.collections.CollectionUtils;
```suggestion
        if (!CollectionUtils.isEmpty(streamExtList)) {
```

**Comment by healchow** [comment]
I don't seem to see unit tests. There are many modifications. It is recommended to add unit tests to key logic.

File: inlong-manager/manager-service/pom.xml [hunk 6]
```diff
@@ -557,5 +557,13 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hudi</groupId>
+            <artifactId>hudi-flink1.13-bundle</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-table-common</artifactId>
+        </dependency>
```
Comment (gong) [inline inlong-manager/manager-service/pom.xml:567]: Why add it in here

Comment (ZuoFengZhang) [reply inlong-manager/manager-service/pom.xml:567]: flink-table-common is used to parse the schema of the hive table, has been rewritten, and removed the reference

**Comment by ZuoFengZhang** [comment]
> I don't seem to see unit tests. There are many modifications. It is recommended to add unit tests to key logic.

I have added the UT about manager-server

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        String inputFormatClassName =
+                HoodieInputFormatUtils.getInputFormatClassName(baseFileFormat, useRealTimeInputFormat);
+        String outputFormatClassName = HoodieInputFormatUtils.getOutputFormatClassName(baseFileFormat);
+        String serDeClassName = HoodieInputFormatUtils.getSerDeClassName(baseFileFormat);
+        sd.setInputFormat(inputFormatClassName);
+        sd.setOutputFormat(outputFormatClassName);
+        Map<String, String> serdeProperties = new HashMap<>();
+        serdeProperties.put("path", location);
+        serdeProperties.put(ConfigUtils.IS_QUERY_AS_RO_TABLE, String.valueOf(!useRealTimeInputFormat));
+        serdeProperties.put("serialization.format", "1");
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java:221]: Why set the `format` to `1`? Does it have any value else?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+        sd.setSerdeInfo(new SerDeInfo(null, serDeClassName, serdeProperties));
+
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java:224]: Unnecessary blank line.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java (hunk 7 above)
Comment (ZuoFengZhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java:221]: `serialization.format` is the field delimiter character in file between two column fields of the table when the file is serialized. This option does not take effect in the hudi table, so can cancel.

**Review by yunqingmoswu — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by leosanqing — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**